src/classes/ADDR_Account_TDTM.cls

```diff
@@ -47,124 +70,183 @@ public class ADDR_Account_TDTM extends TDTM_Runnable {
+                //Putting new address with parent Account in map.
+                newUpdatedAddrsParentAccIDs.add(addr.Parent_Account__c);
+                newUpdatedAddrsParentAccIdToAddr.put(addr.Parent_Account__c, addr);  
+            }
+        }
+        
+        //We need to re-query because we are in the "after" part of the trigger. We cannot change it to before because
+        //we need the Account ID to have a value when populating the addr.Parent_Account__c field above.
+        List<Account> parentAccs = Database.query(ADDR_Addresses_UTIL.getParentAccsWithChildrenQuery() + ':newUpdatedAddrsParentAccIDs');
+                
+        for(Account acc : parentAccs) {
+            Address__c childAddr = newUpdatedAddrsParentAccIdToAddr.get(acc.Id);
+            //uncheck the old default address
+            if(childAddr.Default_Address__c)
+                ADDR_Addresses_UTIL.uncheckDefaultOtherAddrs(childAddr, acc.Addresses__r, dmlWrapper);
+            //link the account to the new default
+            if(childAddr != null) {
+                acc.Current_Address__c = childAddr.Id;
+                dmlWrapper.objectsToUpdate.add(acc);
+            }
+            //copy address info to child contacts, if it's a household and they don't have override
+            if(acc.RecordTypeId == UTIL_Describe.getHhAccRecTypeID()) {
+                for(Contact contact : acc.Contacts) {
+                    if(!contact.is_Address_Override__c) {
+                        ADDR_Addresses_UTIL.copyAddressStdSObj(acc, 'Billing', contact, 'Mailing');
+                        contact.Current_Address__c = childAddr.Id;
+                        dmlWrapper.objectsToUpdate.add(contact);
+                    }
+                }
+            }
+        }
+    }
+    
+    private void addrInfoDeleted(List<Account> accsAddrInfoCleared, DmlWrapper dmlWrapper) {
+        
+        //We need to re-query before we are in the "after" part of the trigger
+        List<Account> accs = [select recordTypeID, Current_Address__c,
+                (select Default_Address__c, Latest_Start_Date__c, Latest_End_Date__c from Account.Addresses__r),
+                (select is_Address_Override__c, MailingStreet, MailingCity, MailingState, MailingPostalCode, MailingCountry, MailingLatitude, MailingLongitude 
+                    from Account.Contacts)
+            from Account where ID in :accsAddrInfoCleared];
+        
+        //Gather the IDs of those addresses that were children of the account
+        Set<ID> oldCurrentAddrIDs = new Set<ID>();
+        for(Account acc : accs) {
+            if(acc.Current_Address__c != null) // get all current addrs
+                oldCurrentAddrIDs.add(acc.Current_Address__c);
+        }
+        
+        for(Account acc : accs) {
+            
+            //Clear Current_Address__c field from accs
+            acc.Current_Address__c = null;
+            dmlWrapper.objectsToUpdate.add(acc);
```

Comment:
won't the accounts passed in from the Run method already be in their afterUpdate?  is it safe to update an object within its AfterUpdate?  Could this have been done on the beforeUpdate to just update this field on the account, rather than a second update?